### PR TITLE
MYNW-121-fix-account-id-overflow-for-nft-modal

### PR DESCRIPTION
**Pull request by NotEternal**
Task: 
https://mnw.atlassian.net/browse/MYNW-121

Preview (attention on the receiver ID):
https://www.loom.com/share/27e6b26527064e1caecbc8e3ca5bb89d
or
https://user-images.githubusercontent.com/61930014/187254485-101716d2-cc79-4efb-9857-fce5b5766c6c.mp4

Changes:
- created a new function to shorten implicit accoun IDs
- IDs in the NFT transfer modal were wrapped with it
- save original account ids in elements' titles:

up1

### CODE REVIEW

File: packages/frontend/src/utils/account.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { shortenAccountId } from './account';
+
+test('It should shorten account ID with correct format', () => {
```
Comment (Patrick1904) [inline packages/frontend/src/utils/account.test.js:3]: Thanks for adding tests 🙏

**Review by Patrick1904 — APPROVED** [review]
LGTM 👍 

@NotEternal there are quite a few other places in the wallet where ellipsis treatment would be nice - sending fungible tokens, etc. It'd be great if you created a follow-up PR to change this in the remaining places too.

File: packages/frontend/src/utils/constants.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ export const ValidatorVersion = {
     [FARMING_VALIDATOR_VERSION]: FARMING_VALIDATOR_VERSION
 };
 
+export const IMPLICINT_ACCOUNT_MAX_LENGTH = 64;
```
Comment (roshaans) [inline packages/frontend/src/utils/constants.js:9]: nit: spelling mistake -> `IMPLICINT` -> `IMPLICIT`

File: packages/frontend/src/utils/account.js [hunk 3]
```diff
@@ -6,4 +7,18 @@ export const validateEmail = (email) => {
...
 export const isImplicitAccount = (accountId) =>
-    accountId && accountId.length === 64 && !accountId.includes('.');
+    accountId && accountId.length === IMPLICINT_ACCOUNT_MAX_LENGTH && !accountId.includes('.');
+
+const ACCOUNT_ID_SEPARATOR = '...';
+
+export const shortenAccountId = (id, startChars = 8, endChars = 8) => {
+    const numOfRemainingChars = startChars + endChars + ACCOUNT_ID_SEPARATOR.length;
+    const isOutOfScope = id.length < numOfRemainingChars || numOfRemainingChars > IMPLICINT_ACCOUNT_MAX_LENGTH;
+    const isPartMissing = startChars < 1 || endChars < 1;
```
Comment (Patrick1904) [inline packages/frontend/src/utils/account.js:17]: Can we rename this to `invalidCharCount`?

**Comment by NotEternal** [comment]
> LGTM +1
> 
> @NotEternal there are quite a few other places in the wallet where ellipsis treatment would be nice - sending fungible tokens, etc. It'd be great if you created a follow-up PR to change this in the remaining places too.

I'll look at those places while learning codebase and add them in a new issue

File: packages/frontend/src/utils/account.test.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { shortenAccountId } from './account';
+
```
Comment (roshaans) [inline packages/frontend/src/utils/account.test.js:2]: Rename file to be something more specific like shortenAccountId.test.js

**Review by roshaans — CHANGES_REQUESTED** [review]
Nice job! :) Just a few nit picky things to change.

File: packages/frontend/src/utils/constants.js [hunk 5]
```diff
@@ -6,7 +6,7 @@ export const ValidatorVersion = {
     [FARMING_VALIDATOR_VERSION]: FARMING_VALIDATOR_VERSION
 };
 
-export const IMPLICINT_ACCOUNT_MAX_LENGTH = 64;
+export const IMPLICIT_ACCOUNT_MAX_LENGTH = 64;
```
Comment (roshaans) [inline packages/frontend/src/utils/constants.js:9]: Nit: Change name from `IMPLICIT_ACCOUNT_MAX_LENGTH` -> `IMPLICIT_ACCOUNT_LENGTH`

**Review by roshaans — APPROVED** [review]
Nice job :) 

It will be nice in later PRs as @Patrick1904 mentioned to introduce this to different parts of the app.  LGTM!

**Pull request merged**